docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,18 @@ a && b()
 a, b()
 
 c = a, b;
+
+(function namedFunctionInExpressionContext () {f();});
```

Comment:
Yeah, I addressed this in my amended commit, though the pairs are not in a corresponding order between the problem/not-a-problem sections (nor had it been so before), mostly because I thought the not-a-problem section should start with easy to follow examples, while the problem section ought to draw out the edge cases first (so people wouldn't be unaware of them).

And yeah, I have a Masters in TESOL/Applied Linguistics, so you were on the mark in using the minimal pair comparison. :)